VirtoCommerce.Storefront/Domain/StaticContent/StaticContentService.cs

```diff
@@ -24,47 +20,47 @@ namespace VirtoCommerce.Storefront.Domain
     /// </summary>
     public class StaticContentService : IStaticContentService
     {
-        private static readonly Regex _headerRegExp = new Regex(@"(?s:^---(.*?)---)");
-        private static readonly string[] _extensions = { ".md", ".liquid", ".html" };
-        private readonly IStorefrontUrlBuilder _urlBuilder;
+        private static readonly string[] _extensions = { ".md", ".liquid", ".html", ".page" };
         private readonly IStaticContentItemFactory _contentItemFactory;
         private readonly IContentBlobProvider _contentBlobProvider;
-        private readonly MarkdownPipeline _markdownPipeline;
         private readonly IStorefrontMemoryCache _memoryCache;
+        private readonly IStaticContentLoaderFactory _metadataFactory;
+        private readonly IApiChangesWatcher _apiChangesWatcher;
         private readonly string _basePath = "Pages";
 
-        public StaticContentService(IStorefrontMemoryCache memoryCache, IWorkContextAccessor workContextAccessor,
-                                        IStorefrontUrlBuilder urlBuilder, IStaticContentItemFactory contentItemFactory,
-                                        IContentBlobProvider contentBlobProvider)
+        public StaticContentService(IStorefrontMemoryCache memoryCache, IStaticContentItemFactory contentItemFactory,
+                                        IContentBlobProvider contentBlobProvider, IStaticContentLoaderFactory metadataFactory, IApiChangesWatcher changesWatcher)
         {
-            _urlBuilder = urlBuilder;
             _contentItemFactory = contentItemFactory;
             _contentBlobProvider = contentBlobProvider;
             _memoryCache = memoryCache;
-            _markdownPipeline = new MarkdownPipelineBuilder().UseAdvancedExtensions().Build();
+            _metadataFactory = metadataFactory;
+            _apiChangesWatcher = changesWatcher;
         }
 
         #region IStaticContentService Members
 
         public IEnumerable<ContentItem> LoadStoreStaticContent(Store store)
         {
-            var baseStoreContentPath = _basePath + "/" + store.Id;
+            var baseStoreContentPath = string.Concat(_basePath, "/", store.Id);
             var cacheKey = CacheKey.With(GetType(), "LoadStoreStaticContent", store.Id);
             return _memoryCache.GetOrCreateExclusive(cacheKey, (cacheEntry) =>
             {
                 cacheEntry.AddExpirationToken(new CompositeChangeToken(new[] { StaticContentCacheRegion.CreateChangeToken(), _contentBlobProvider.Watch(baseStoreContentPath + "/**/*") }));
+                cacheEntry.AddExpirationToken(_apiChangesWatcher.CreateChangeToken());
```

Comment:
this line will lead to the ineffective working of cache thus any changes with data will expire all static content in the cache. 
This is absolutely unexpected behavior, why do we evict the static pages from the cache when changing a product or price, for example?